examples/jsm/nodes/materials/NodeMaterial.js

```diff
@@ -61,17 +61,11 @@ Object.defineProperties( NodeMaterial.prototype, {
 
 NodeMaterial.prototype.onBeforeCompile = function ( shader, renderer ) {
 
-	var materialProperties = renderer.properties.get( this );
+	this.build( { renderer: renderer } );
 
-	if ( this.version !== materialProperties.__version ) {
```

Comment:
Hmm, what did this code check?

Reply:
Since `Material.onBeforeCompile()` was previously called way more often than now, I suspect this check should prevent unnecessary material builds. However, just checking for `materialProperties.__version` is not correct anyway since program updates might be necessary because of other reasons (e.g. `WebGLRenderer.outputEncoding` changes). 

Hence, it's more correct if the material system let `WebGLRenderer`  decide _when_ to update shaders.